modules/core/m_identified.c

```diff
@@ -0,0 +1,64 @@
+#include <stdinc.h>
+#include <modules.h>
+#include <messages.h>
+#include <send.h>
+
+static const char identified_desc[] = "Provides the IDENTIFIED server-to-server command";
+
+static void m_identified(struct MsgBuf *, struct Client *, struct Client *, int, const char *[]);
+
+static struct Message identified_msgtab = {
+	"IDENTIFIED", 0, 0, 0, 0,
+	{mg_ignore, mg_ignore, mg_ignore, mg_ignore, {m_identified, 3}, mg_ignore}
+};
+
+static mapi_clist_av1 identified_clist[] = {
+	&identified_msgtab,
+	NULL
+};
+
+static void identified_nick_change(void *);
+static void identified_burst_client(void *);
+
+static mapi_hfn_list_av1 identified_hfnlist[] = {
+	{ "local_nick_change", identified_nick_change, HOOK_MONITOR },
+	{ "remote_nick_change", identified_nick_change, HOOK_MONITOR },
+	{ "burst_client", identified_burst_client },
+	{ NULL, NULL }
+};
+
+static void m_identified(struct MsgBuf *msgbuf, struct Client *client_p, struct Client *source_p, int parc, const char *parv[])
+{
+	struct Client *target_p = find_person(parv[1]);
+	const char *nick = parv[2];
+
+	if (target_p == NULL)
+		return;
+
+	if (irccmp(target_p->name, nick))
```

Comment:
If this UID is _not_ using the nickname that services says its identified for, should this set this client to being unidentified?

Reply:
I don't think it matters much. If they're not using the right nickname because they changed nick, they've already been unidentified, and that's the only case I can think of.